src/classic/clvm_tools/stages/stage_2/module.rs

```diff
@@ -554,27 +561,42 @@ fn compile_functions(
     macro_lookup_program: NodePtr,
     constants_symbol_table: &[(NodePtr, Vec<u8>)],
     args_root_node: &NodePath,
-) -> Result<HashMap<Vec<u8>, NodePtr>, EvalErr> {
-    let compiled_functions = HashMap::new();
+    has_constants_tree: bool,
+) -> Result<CompileOutput, EvalErr> {
+    let compiled = Default::default();
 
     return fold_m(
         allocator,
-        &|allocator: &mut Allocator, compiled_functions, name_exp: (&Vec<u8>, &NodePtr)| {
+        &|allocator: &mut Allocator, compiled, name_exp: (&Vec<u8>, &NodePtr)| {
             add_one_function(
                 allocator,
                 args_root_node,
                 macro_lookup_program,
                 constants_symbol_table,
-                compiled_functions,
+                compiled,
                 name_exp.0,
                 *name_exp.1,
+                has_constants_tree,
             )
         },
-        compiled_functions,
+        compiled,
         &mut functions.iter(),
     );
 }
 
+// Prepend __chia__main_arguments with the indicated argument list.
```

Comment:
Do you mean one of the following?

`Prepend __chia__main_arguments to the indicated argument list.`

`Append the indicated argument list to __chia__main_arguments.`

Reply:
Add an entry for main's arguments, named __chia__main_arguments in the symbols, to the symbol list, placing it at the front for simplicity.